tests/integration/nm/ipv4_test.py

```diff
@@ -50,6 +50,19 @@ def test_interface_ipv4_change(eth1_up):
     assert ip4_expected_state == ipv4_current_state
 
 
+def test_activate_dhcp_on_existing_interface(eth1_up):
+    expected_ipv4_state = {'enabled': True, 'dhcp': True, 'address': []}
+    with mainloop():
+        _modify_interface(ipv4_state=expected_ipv4_state)
+
+    nm.nmclient.client(refresh=True)
+    ipv4_current_state = _get_ipv4_current_state(TEST_IFACE)
+    assert ipv4_current_state == dict(
+        expected_ipv4_state,
+        **{'auto-dns': True, 'auto-gateway': True, 'auto-routes': True}
```

Comment:
This is a bit strange. You could prepare the full expected state in advance in one shot.
I think that the problem is the attempt to share the "expected" state between the desired and expect.
But even if you decide to keep it, you can just update the expected using `update` one line before.